### ISSUE TITLE
feat: sklearn-interface NN estimator with categorical embeddings

### ISSUE THREAD

**Issue by sunkusun9** [sunkusun9/ml-labs#82]
## Overview

Implement a sklearn-compatible neural network estimator that handles categorical variables via entity embeddings. TensorFlow-first implementation, with PyTorch version to follow.

## Motivation

Gradient boosting models (LGB/XGB/CatBoost) are already well-supported via adapters. Adding a NN estimator that natively handles categorical embeddings completes the modeling toolkit for tabular data.

### Embedding size

Auto-rule (default): `min(50, (cardinality + 1) // 2)`

Per-column override via dict: `{"color": 3, "city": 20}` — more intuitive than pytorch-tabular's `(cardinality, dim)` tuple list.

### Input construction

Accepts pandas/polars/numpy input. Internally splits columns into:
- categorical inputs → integer-encoded → individual `Embedding` layers
- continuous inputs → single dense input

Embedding outputs are concatenated with continuous features before entering the FF network.

### sklearn interface

- `NNClassifier(BaseEstimator, ClassifierMixin)` / `NNRegressor(BaseEstimator, RegressorMixin)`
- `fit(X, y)`, `predict(X)`, `predict_proba(X)` (classifier)
- `get_params()` / `set_params()` for Optuna / GridSearchCV compatibility

### Key parameters

```python
NNClassifier(
    cat_cols=None,              # list of categorical column names (or indices)
    embedding_dims=None,        # None=auto, or dict {col: dim}
    layers=[256, 128],          # hidden layer sizes
    dropout=0.3,
    activation="relu",
    epochs=100,
    batch_size=1024,
    learning_rate=1e-3,
    validation_fraction=0.1,    # internal early stopping
    callbacks=None,             # pass-through to tf.keras
)
```

### Data format handling

- Label-encode categoricals on `fit`, apply same encoding on `transform`/`predict`
- Handle unseen categories (OOV) gracefully (map to 0 / dedicated OOV index)
- Support pandas Categorical dtype for auto-detecting cat_cols

## Implementation Plan

1. `mllabs/processor/_nn_estimator.py` — core estimator
2. TensorFlow backend first (`_nn_tf.py`)
3. PyTorch backend later (`_nn_torch.py`), same interface
4. Export via `mllabs/processor/__init__.py` under optional import guard

## References

- pytorch-tabular `CategoryEmbeddingModel` — embedding auto-size rule
- fastai entity embeddings — original heuristic
- embedding-encoder — sklearn-compatible reference implementation
